circle.yml

```diff
@@ -1,35 +1,84 @@
+defaults: &defaults
+  working_directory: ~/repo
+  docker:
+    - image: applicaster/zapp:0.8
+
+whitelist: &whitelist
+  paths:
+    - node_modules/*
+    - src/*
+    - LICENSE
+    - package.json
+    - package-lock.json
+    - README.md
+    - .eslintrc
+    - .babelrc
+
 version: 2
 jobs:
   build:
-    docker:
-      - image: applicaster/zapp:0.8
+    <<: *defaults
     steps:
       - restore_cache:
           key: react-native-zapp-bridge-{{ .Branch }}-{{ .Revision }}
```

Comment:
I would add version like`-v1`, without it, it will be less intuitive to invalidate cache

Reply:
I'm not sure I understand;
`react-native-zapp-bridge-{{ .Branch }}-{{ .Revision }}-v1`?
How does it make it more intuitive?

Reply:
In ci 2.0 the cache mechanism weirdly ignores ‘rebuild without cache’
So if you change this file to check a cache with `v2` key, it won’t find it, so we have a reliable way to invalidate cache on a branch / commit

Reply:
https://user-images.githubusercontent.com/5968910/37973854-20e465ae-31e4-11e8-9db1-21e167dd717b.png

Reply:
checkout circle ci docs and specific tip:
https://circleci.com/docs/2.0/caching/#clearing-cache